{
  "SchemaVersion": 2,
  "CreatedAt": "2025-09-30T01:08:57.349128+03:00",
  "ArtifactName": "ghcr.io/igorgolm/aws-cd-georgia-demo-app:f705934d36bd1fa4baebbb8125382424b4281497",
  "ArtifactType": "container_image",
  "Metadata": {
    "OS": {
      "Family": "debian",
      "Name": "13.1"
    },
    "ImageID": "sha256:d68a9a326782c8fa166c32ca22841a05b33edaccc2099b958bfc3032825202f4",
    "DiffIDs": [
      "sha256:daf557c4f08e0cb77b2af2a08d8bca4f333cf408d88191911715e6f393eca215",
      "sha256:c0cf2e680f47fb14f14dbf373bd2bca6edfcee9ea05b97c2ba5af79b3fd4a03c",
      "sha256:585b35c14bc730eddf590406ab5bcebc3ac34a4b22f3825558fc375115020f02",
      "sha256:7dfdd45451be2899a4784597610e93bfa6b6172a7a7e775397c3aac6d1436e12",
      "sha256:708103fa83c319e4a5e397a55f16b6e6760961773b3f5e8d336ba021188d85dc",
      "sha256:332d186351dc36cd122db694e3c489197006ae48f849e3ccc333020eb6f1e233",
      "sha256:31e65093e97e693a07afbccbd23733442dd2d1e3bef63cf09ca361fe47b4eff0",
      "sha256:b590a81b7631ed84268b8c2cb73c621dfdc1d2a6351d48820c4bc248177cc6f0",
      "sha256:a9779c2a2f73f128d8363871f1cfd0fff1be83ba7e716ac05b97393490dc47e2",
      "sha256:6f7f0fad65ba2b550b90f714ed8f78bd55ce3666488eb8afa2c198952e902bb6",
      "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e",
      "sha256:a8138d9b8da1db8814d5449bd75f31ac279778f4cca36acb6609bec81b2fb2e1"
    ],
    "RepoTags": [
      "ghcr.io/igorgolm/aws-cd-georgia-demo-app:f705934d36bd1fa4baebbb8125382424b4281497"
    ],
    "RepoDigests": [
      "ghcr.io/igorgolm/aws-cd-georgia-demo-app@sha256:670bf37a3941e1508a4d1cdf6ea7e7ad41a2fd4a0865fc6aee6c707c87e8fb07"
    ],
    "ImageConfig": {
      "architecture": "amd64",
      "created": "2025-09-29T21:17:08.698519388Z",
      "history": [
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "# debian.sh --arch 'amd64' out/ 'trixie' '@1757289600'",
          "comment": "debuerreotype 0.16"
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "ENV PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "ENV LANG=C.UTF-8",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "RUN /bin/sh -c set -eux; \tapt-get update; \tapt-get install -y --no-install-recommends \t\tca-certificates \t\tnetbase \t\ttzdata \t; \tapt-get dist-clean # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "ENV GPG_KEY=7169605F62C751356D054A26A821E680E5FA6305",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "ENV PYTHON_VERSION=3.12.11",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "ENV PYTHON_SHA256=c30bb24b7f1e9a19b11b55a546434f74e739bb4c271a3e3a80ff4380d49f7adb",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "RUN /bin/sh -c set -eux; \t\tsavedAptMark=\"$(apt-mark showmanual)\"; \tapt-get update; \tapt-get install -y --no-install-recommends \t\tdpkg-dev \t\tgcc \t\tgnupg \t\tlibbluetooth-dev \t\tlibbz2-dev \t\tlibc6-dev \t\tlibdb-dev \t\tlibffi-dev \t\tlibgdbm-dev \t\tliblzma-dev \t\tlibncursesw5-dev \t\tlibreadline-dev \t\tlibsqlite3-dev \t\tlibssl-dev \t\tmake \t\ttk-dev \t\tuuid-dev \t\twget \t\txz-utils \t\tzlib1g-dev \t; \t\twget -O python.tar.xz \"https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz\"; \techo \"$PYTHON_SHA256 *python.tar.xz\" | sha256sum -c -; \twget -O python.tar.xz.asc \"https://www.python.org/ftp/python/${PYTHON_VERSION%%[a-z]*}/Python-$PYTHON_VERSION.tar.xz.asc\"; \tGNUPGHOME=\"$(mktemp -d)\"; export GNUPGHOME; \tgpg --batch --keyserver hkps://keys.openpgp.org --recv-keys \"$GPG_KEY\"; \tgpg --batch --verify python.tar.xz.asc python.tar.xz; \tgpgconf --kill all; \trm -rf \"$GNUPGHOME\" python.tar.xz.asc; \tmkdir -p /usr/src/python; \ttar --extract --directory /usr/src/python --strip-components=1 --file python.tar.xz; \trm python.tar.xz; \t\tcd /usr/src/python; \tgnuArch=\"$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)\"; \t./configure \t\t--build=\"$gnuArch\" \t\t--enable-loadable-sqlite-extensions \t\t--enable-optimizations \t\t--enable-option-checking=fatal \t\t--enable-shared \t\t$(test \"${gnuArch%%-*}\" != 'riscv64' \u0026\u0026 echo '--with-lto') \t\t--with-ensurepip \t; \tnproc=\"$(nproc)\"; \tEXTRA_CFLAGS=\"$(dpkg-buildflags --get CFLAGS)\"; \tLDFLAGS=\"$(dpkg-buildflags --get LDFLAGS)\"; \tLDFLAGS=\"${LDFLAGS:--Wl},--strip-all\"; \t\tarch=\"$(dpkg --print-architecture)\"; arch=\"${arch##*-}\"; \t\tcase \"$arch\" in \t\t\tamd64|arm64) \t\t\t\tEXTRA_CFLAGS=\"${EXTRA_CFLAGS:-} -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer\"; \t\t\t\t;; \t\t\ti386) \t\t\t\t;; \t\t\t*) \t\t\t\tEXTRA_CFLAGS=\"${EXTRA_CFLAGS:-} -fno-omit-frame-pointer\"; \t\t\t\t;; \t\tesac; \tmake -j \"$nproc\" \t\t\"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}\" \t\t\"LDFLAGS=${LDFLAGS:-}\" \t; \trm python; \tmake -j \"$nproc\" \t\t\"EXTRA_CFLAGS=${EXTRA_CFLAGS:-}\" \t\t\"LDFLAGS=${LDFLAGS:--Wl},-rpath='\\$\\$ORIGIN/../lib'\" \t\tpython \t; \tmake install; \t\tcd /; \trm -rf /usr/src/python; \t\tfind /usr/local -depth \t\t\\( \t\t\t\\( -type d -a \\( -name test -o -name tests -o -name idle_test \\) \\) \t\t\t-o \\( -type f -a \\( -name '*.pyc' -o -name '*.pyo' -o -name 'libpython*.a' \\) \\) \t\t\\) -exec rm -rf '{}' + \t; \t\tldconfig; \t\tapt-mark auto '.*' \u003e /dev/null; \tapt-mark manual $savedAptMark; \tfind /usr/local -type f -executable -not \\( -name '*tkinter*' \\) -exec ldd '{}' ';' \t\t| awk '/=\u003e/ { so = $(NF-1); if (index(so, \"/usr/local/\") == 1) { next }; gsub(\"^/(usr/)?\", \"\", so); printf \"*%s\\n\", so }' \t\t| sort -u \t\t| xargs -rt dpkg-query --search \t\t| awk 'sub(\":$\", \"\", $1) { print $1 }' \t\t| sort -u \t\t| xargs -r apt-mark manual \t; \tapt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \tapt-get dist-clean; \t\texport PYTHONDONTWRITEBYTECODE=1; \tpython3 --version; \tpip3 --version # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "RUN /bin/sh -c set -eux; \tfor src in idle3 pip3 pydoc3 python3 python3-config; do \t\tdst=\"$(echo \"$src\" | tr -d 3)\"; \t\t[ -s \"/usr/local/bin/$src\" ]; \t\t[ ! -e \"/usr/local/bin/$dst\" ]; \t\tln -svT \"$src\" \"/usr/local/bin/$dst\"; \tdone # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-08-08T18:20:34Z",
          "created_by": "CMD [\"python3\"]",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-09-29T21:16:29.736169245Z",
          "created_by": "ENV PYTHONDONTWRITEBYTECODE=1 PYTHONUNBUFFERED=1 PORT=8080 GUNICORN_WORKERS=2 GUNICORN_THREADS=8",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-09-29T21:16:29.736169245Z",
          "created_by": "WORKDIR /app",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:16:32.215608369Z",
          "created_by": "RUN /bin/sh -c pip install --upgrade pip # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:16:42.433156288Z",
          "created_by": "RUN /bin/sh -c apt-get update \u0026\u0026 apt-get install -y --no-install-recommends     build-essential     gcc     zlib1g-dev     libjpeg62-turbo-dev     libpng-dev     libtiff5-dev \u0026\u0026     rm -rf /var/lib/apt/lists/* # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:16:42.490127841Z",
          "created_by": "COPY requirements.txt ./ # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:16:44.297150987Z",
          "created_by": "RUN /bin/sh -c pip install --no-cache-dir -r requirements.txt # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:16:44.31649795Z",
          "created_by": "COPY . . # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:17:08.617835639Z",
          "created_by": "RUN /bin/sh -c pip install --no-cache-dir \"Pillow==9.0.0\" \"urllib3==1.25.8\" # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:17:08.698519388Z",
          "created_by": "RUN /bin/sh -c useradd -m appuser \u0026\u0026 chown -R appuser:appuser /app # buildkit",
          "comment": "buildkit.dockerfile.v0"
        },
        {
          "created": "2025-09-29T21:17:08.698519388Z",
          "created_by": "USER appuser",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-09-29T21:17:08.698519388Z",
          "created_by": "EXPOSE map[8080/tcp:{}]",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        },
        {
          "created": "2025-09-29T21:17:08.698519388Z",
          "created_by": "CMD [\"sh\" \"-lc\" \"gunicorn -w ${GUNICORN_WORKERS} -k gthread --threads ${GUNICORN_THREADS} -b 0.0.0.0:${PORT} app:app\"]",
          "comment": "buildkit.dockerfile.v0",
          "empty_layer": true
        }
      ],
      "os": "linux",
      "rootfs": {
        "type": "layers",
        "diff_ids": [
          "sha256:daf557c4f08e0cb77b2af2a08d8bca4f333cf408d88191911715e6f393eca215",
          "sha256:c0cf2e680f47fb14f14dbf373bd2bca6edfcee9ea05b97c2ba5af79b3fd4a03c",
          "sha256:585b35c14bc730eddf590406ab5bcebc3ac34a4b22f3825558fc375115020f02",
          "sha256:7dfdd45451be2899a4784597610e93bfa6b6172a7a7e775397c3aac6d1436e12",
          "sha256:708103fa83c319e4a5e397a55f16b6e6760961773b3f5e8d336ba021188d85dc",
          "sha256:332d186351dc36cd122db694e3c489197006ae48f849e3ccc333020eb6f1e233",
          "sha256:31e65093e97e693a07afbccbd23733442dd2d1e3bef63cf09ca361fe47b4eff0",
          "sha256:b590a81b7631ed84268b8c2cb73c621dfdc1d2a6351d48820c4bc248177cc6f0",
          "sha256:a9779c2a2f73f128d8363871f1cfd0fff1be83ba7e716ac05b97393490dc47e2",
          "sha256:6f7f0fad65ba2b550b90f714ed8f78bd55ce3666488eb8afa2c198952e902bb6",
          "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e",
          "sha256:a8138d9b8da1db8814d5449bd75f31ac279778f4cca36acb6609bec81b2fb2e1"
        ]
      },
      "config": {
        "Cmd": [
          "sh",
          "-lc",
          "gunicorn -w ${GUNICORN_WORKERS} -k gthread --threads ${GUNICORN_THREADS} -b 0.0.0.0:${PORT} app:app"
        ],
        "Env": [
          "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
          "LANG=C.UTF-8",
          "GPG_KEY=7169605F62C751356D054A26A821E680E5FA6305",
          "PYTHON_VERSION=3.12.11",
          "PYTHON_SHA256=c30bb24b7f1e9a19b11b55a546434f74e739bb4c271a3e3a80ff4380d49f7adb",
          "PYTHONDONTWRITEBYTECODE=1",
          "PYTHONUNBUFFERED=1",
          "PORT=8080",
          "GUNICORN_WORKERS=2",
          "GUNICORN_THREADS=8"
        ],
        "User": "appuser",
        "WorkingDir": "/app",
        "ExposedPorts": {
          "8080/tcp": {}
        },
        "ArgsEscaped": true
      }
    }
  },
  "Results": [
    {
      "Target": "ghcr.io/igorgolm/aws-cd-georgia-demo-app:f705934d36bd1fa4baebbb8125382424b4281497 (debian 13.1)",
      "Class": "os-pkgs",
      "Type": "debian"
    },
    {
      "Target": "Python",
      "Class": "lang-pkgs",
      "Type": "python-pkg",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2022-22817",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "9.0.1",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-22817",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-pillow: PIL.ImageMath.eval allows evaluation of arbitrary expressions",
          "Description": "PIL.ImageMath.eval in Pillow before 9.0.0 allows evaluation of arbitrary expressions, such as ones that use the Python exec method. A lambda expression could also be used.",
          "Severity": "CRITICAL",
          "VendorSeverity": {
            "alma": 3,
            "amazon": 3,
            "bitnami": 4,
            "ghsa": 4,
            "nvd": 4,
            "oracle-oval": 3,
            "redhat": 3,
            "rocky": 3,
            "ubuntu": 2
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 9.8
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 9.8
            },
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V2Score": 7.5,
              "V3Score": 9.8
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 9.8
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2022-22817",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2042522",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2042527",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-22815",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-22816",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-22817",
            "https://errata.almalinux.org/8/ALSA-2022-0643.html",
            "https://errata.rockylinux.org/RLSA-2022:0643",
            "https://github.com/advisories/GHSA-8vj2-vxx3-667w",
            "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2022-10.yaml",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/commit/8531b01d6cdf0b70f256f93092caa2a5d91afc11",
            "https://linux.oracle.com/cve/CVE-2022-22817.html",
            "https://linux.oracle.com/errata/ELSA-2022-0643.html",
            "https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html",
            "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2022-22817",
            "https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling",
            "https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#restrict-builtins-available-to-imagemath-eval",
            "https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security",
            "https://security.gentoo.org/glsa/202211-10",
            "https://ubuntu.com/security/notices/USN-5227-1",
            "https://ubuntu.com/security/notices/USN-5227-2",
            "https://ubuntu.com/security/notices/USN-5227-3",
            "https://www.cve.org/CVERecord?id=CVE-2022-22817",
            "https://www.debian.org/security/2022/dsa-5053"
          ],
          "PublishedDate": "2022-01-10T14:12:55.16Z",
          "LastModifiedDate": "2024-11-21T06:47:30.477Z"
        },
        {
          "VulnerabilityID": "CVE-2023-50447",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "10.2.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-50447",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "pillow: Arbitrary Code Execution via the environment parameter",
          "Description": "Pillow through 10.1.0 allows PIL.ImageMath.eval Arbitrary Code Execution via the environment parameter, a different vulnerability than CVE-2022-22817 (which was about the expression parameter).",
          "Severity": "CRITICAL",
          "CweIDs": [
            "CWE-94",
            "CWE-95"
          ],
          "VendorSeverity": {
            "alma": 3,
            "amazon": 3,
            "bitnami": 3,
            "ghsa": 4,
            "nvd": 3,
            "oracle-oval": 3,
            "redhat": 3,
            "ubuntu": 2
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 8.1
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 8.1
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 8.1
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 8.1
            }
          },
          "References": [
            "http://www.openwall.com/lists/oss-security/2024/01/20/1",
            "https://access.redhat.com/errata/RHSA-2024:0893",
            "https://access.redhat.com/security/cve/CVE-2023-50447",
            "https://bugzilla.redhat.com/2259479",
            "https://devhub.checkmarx.com/cve-details/CVE-2023-50447",
            "https://devhub.checkmarx.com/cve-details/CVE-2023-50447/",
            "https://duartecsantos.github.io/2023-01-02-CVE-2023-50447",
            "https://duartecsantos.github.io/2023-01-02-CVE-2023-50447/",
            "https://duartecsantos.github.io/2024-01-02-CVE-2023-50447",
            "https://duartecsantos.github.io/2024-01-02-CVE-2023-50447/",
            "https://errata.almalinux.org/8/ALSA-2024-0893.html",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/commit/45c726fd4daa63236a8f3653530f297dc87b160a",
            "https://github.com/python-pillow/Pillow/releases",
            "https://linux.oracle.com/cve/CVE-2023-50447.html",
            "https://linux.oracle.com/errata/ELSA-2024-0893.html",
            "https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-50447",
            "https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#imagemath-eval-restricted-environment-keys",
            "https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#security",
            "https://ubuntu.com/security/notices/USN-6618-1",
            "https://www.cve.org/CVERecord?id=CVE-2023-50447"
          ],
          "PublishedDate": "2024-01-19T20:15:11.87Z",
          "LastModifiedDate": "2024-11-21T08:37:00.967Z"
        },
        {
          "VulnerabilityID": "CVE-2022-24303",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "9.0.1",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-24303",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-pillow: temporary directory with a space character allows removal of unrelated file after im.show() and related actions",
          "Description": "Pillow before 9.0.1 allows attackers to delete files because spaces in temporary pathnames are mishandled.",
          "Severity": "HIGH",
          "VendorSeverity": {
            "bitnami": 4,
            "ghsa": 3,
            "nvd": 4,
            "redhat": 2,
            "ubuntu": 1
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:H",
              "V3Score": 9.1
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:H",
              "V3Score": 9.1
            },
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:N/I:P/A:P",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:H",
              "V2Score": 6.4,
              "V3Score": 9.1
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:H/A:H",
              "V3Score": 7.4
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2022-24303",
            "https://github.com/advisories/GHSA-9j59-75qj-795w",
            "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2022-168.yaml",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/blob/e8ab5640774716c5486d3cb05167f74f742ad6ef/CHANGES.rst?plain=1#L1172",
            "https://github.com/python-pillow/Pillow/commit/10c4f75aaa383bd9671e923e3b91d391ea12d781",
            "https://github.com/python-pillow/Pillow/commit/143032103c9f2d55a0a7960bd3e630cb72549e8a",
            "https://github.com/python-pillow/Pillow/commit/427221ef5f19157001bf8b1ad7cfe0b905ca8c26",
            "https://github.com/python-pillow/Pillow/pull/3450",
            "https://github.com/python-pillow/Pillow/pull/6010",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/W4ZUXPKEX72O3E5IHBPVY5ZCPMJ4GHHV/",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XR6UP2XONXOVXI4446VY72R63YRO2YTP/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/W4ZUXPKEX72O3E5IHBPVY5ZCPMJ4GHHV",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XR6UP2XONXOVXI4446VY72R63YRO2YTP",
            "https://nvd.nist.gov/vuln/detail/CVE-2022-24303",
            "https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html",
            "https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security",
            "https://security.gentoo.org/glsa/202211-10",
            "https://ubuntu.com/security/notices/USN-5777-1",
            "https://ubuntu.com/security/notices/USN-5777-2",
            "https://www.cve.org/CVERecord?id=CVE-2022-24303"
          ],
          "PublishedDate": "2022-03-28T02:15:07.14Z",
          "LastModifiedDate": "2024-11-21T06:50:07.9Z"
        },
        {
          "VulnerabilityID": "CVE-2022-45198",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "9.2.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-45198",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "Pillow before 9.2.0 performs Improper Handling of Highly Compressed GI ...",
          "Description": "Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).",
          "Severity": "HIGH",
          "VendorSeverity": {
            "amazon": 3,
            "bitnami": 3,
            "ghsa": 3,
            "nvd": 3,
            "ubuntu": 1
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://bugs.gentoo.org/855683",
            "https://cwe.mitre.org/data/definitions/409.html",
            "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2022-42979.yaml",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4",
            "https://github.com/python-pillow/Pillow/pull/6402",
            "https://github.com/python-pillow/Pillow/pull/6402/commits/c9f1b35e981075110a23487a8d4a6cbb59a588ea",
            "https://github.com/python-pillow/Pillow/releases/tag/9.2.0",
            "https://nvd.nist.gov/vuln/detail/CVE-2022-45198",
            "https://security.gentoo.org/glsa/202211-10",
            "https://ubuntu.com/security/notices/USN-5777-1",
            "https://ubuntu.com/security/notices/USN-5777-2",
            "https://www.cve.org/CVERecord?id=CVE-2022-45198"
          ],
          "PublishedDate": "2022-11-14T07:15:10.347Z",
          "LastModifiedDate": "2024-11-21T07:28:56.987Z"
        },
        {
          "VulnerabilityID": "CVE-2023-44271",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "10.0.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-44271",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-pillow: uncontrolled resource consumption when textlength in an ImageDraw instance operates on a long text argument",
          "Description": "An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-770"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "bitnami": 3,
            "ghsa": 3,
            "nvd": 3,
            "oracle-oval": 2,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 1
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:3005",
            "https://access.redhat.com/security/cve/CVE-2023-44271",
            "https://bugzilla.redhat.com/2247820",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2247820",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-44271",
            "https://devhub.checkmarx.com/cve-details/CVE-2023-44271",
            "https://devhub.checkmarx.com/cve-details/CVE-2023-44271/",
            "https://errata.almalinux.org/8/ALSA-2024-3005.html",
            "https://errata.rockylinux.org/RLSA-2024:3005",
            "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2023-227.yaml",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7",
            "https://github.com/python-pillow/Pillow/pull/7244",
            "https://linux.oracle.com/cve/CVE-2023-44271.html",
            "https://linux.oracle.com/errata/ELSA-2024-3005.html",
            "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-44271",
            "https://ubuntu.com/security/notices/USN-6618-1",
            "https://www.cve.org/CVERecord?id=CVE-2023-44271"
          ],
          "PublishedDate": "2023-11-03T05:15:30.137Z",
          "LastModifiedDate": "2024-11-21T08:25:33.61Z"
        },
        {
          "VulnerabilityID": "CVE-2023-4863",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "10.0.1",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-4863",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "libwebp: Heap buffer overflow in WebP Codec",
          "Description": "Heap buffer overflow in libwebp in Google Chrome prior to 116.0.5845.187 and libwebp 1.3.2 allowed a remote attacker to perform an out of bounds memory write via a crafted HTML page. (Chromium security severity: Critical)",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-787"
          ],
          "VendorSeverity": {
            "alma": 3,
            "amazon": 3,
            "cbl-mariner": 3,
            "ghsa": 3,
            "nvd": 3,
            "oracle-oval": 3,
            "photon": 3,
            "redhat": 3,
            "rocky": 3,
            "ubuntu": 2
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 8.8
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 8.8
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H",
              "V3Score": 9.6
            }
          },
          "References": [
            "http://www.openwall.com/lists/oss-security/2023/09/21/4",
            "http://www.openwall.com/lists/oss-security/2023/09/22/1",
            "http://www.openwall.com/lists/oss-security/2023/09/22/3",
            "http://www.openwall.com/lists/oss-security/2023/09/22/4",
            "http://www.openwall.com/lists/oss-security/2023/09/22/5",
            "http://www.openwall.com/lists/oss-security/2023/09/22/6",
            "http://www.openwall.com/lists/oss-security/2023/09/22/7",
            "http://www.openwall.com/lists/oss-security/2023/09/22/8",
            "http://www.openwall.com/lists/oss-security/2023/09/26/1",
            "http://www.openwall.com/lists/oss-security/2023/09/26/7",
            "http://www.openwall.com/lists/oss-security/2023/09/28/1",
            "http://www.openwall.com/lists/oss-security/2023/09/28/2",
            "http://www.openwall.com/lists/oss-security/2023/09/28/4",
            "https://access.redhat.com/errata/RHSA-2023:5224",
            "https://access.redhat.com/security/cve/CVE-2023-4863",
            "https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway",
            "https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/",
            "https://blog.isosceles.com/the-webp-0day",
            "https://blog.isosceles.com/the-webp-0day/",
            "https://bugzilla.redhat.com/2238431",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2238431",
            "https://bugzilla.suse.com/show_bug.cgi?id=1215231",
            "https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html",
            "https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_12.html",
            "https://chromium.googlesource.com/webm/libwebp.git/+/902bc9190331343b2017211debcec8d2ab87e17a",
            "https://crbug.com/1479274",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-4863",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-5129",
            "https://en.bandisoft.com/honeyview/history",
            "https://en.bandisoft.com/honeyview/history/",
            "https://errata.almalinux.org/9/ALSA-2023-5224.html",
            "https://errata.rockylinux.org/RLSA-2023:5214",
            "https://github.com/ImageMagick/ImageMagick/discussions/6664",
            "https://github.com/dlemstra/Magick.NET/releases/tag/13.3.0",
            "https://github.com/electron/electron/pull/39823",
            "https://github.com/electron/electron/pull/39825",
            "https://github.com/electron/electron/pull/39826",
            "https://github.com/electron/electron/pull/39827",
            "https://github.com/electron/electron/pull/39828",
            "https://github.com/jaredforth/webp/commit/9d4c56e63abecc777df71c702503c3eaabd7dcbc",
            "https://github.com/jaredforth/webp/pull/30",
            "https://github.com/python-pillow/Pillow/pull/7395",
            "https://github.com/qnighy/libwebp-sys2-rs/commit/4560c473a76ec8bd8c650f19ddf9d7a44f719f8b",
            "https://github.com/qnighy/libwebp-sys2-rs/pull/21",
            "https://github.com/webmproject/libwebp",
            "https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a",
            "https://github.com/webmproject/libwebp/releases/tag/v1.3.2",
            "https://linux.oracle.com/cve/CVE-2023-4863.html",
            "https://linux.oracle.com/errata/ELSA-2023-5309.html",
            "https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html",
            "https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html",
            "https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/",
            "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863",
            "https://news.ycombinator.com/item?id=37478403",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-4863",
            "https://pillow.readthedocs.io/en/stable/releasenotes/10.0.1.html#security",
            "https://rustsec.org/advisories/RUSTSEC-2023-0060.html",
            "https://rustsec.org/advisories/RUSTSEC-2023-0061.html",
            "https://security-tracker.debian.org/tracker/CVE-2023-4863",
            "https://security.gentoo.org/glsa/202309-05",
            "https://security.gentoo.org/glsa/202401-10",
            "https://security.netapp.com/advisory/ntap-20230929-0011",
            "https://security.netapp.com/advisory/ntap-20230929-0011/",
            "https://sethmlarson.dev/security-developer-in-residence-weekly-report-16",
            "https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863",
            "https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/",
            "https://ubuntu.com/security/notices/USN-6367-1",
            "https://ubuntu.com/security/notices/USN-6368-1",
            "https://ubuntu.com/security/notices/USN-6369-1",
            "https://ubuntu.com/security/notices/USN-6369-2",
            "https://www.bentley.com/advisories/be-2023-0001",
            "https://www.bentley.com/advisories/be-2023-0001/",
            "https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks",
            "https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/",
            "https://www.cisa.gov/known-exploited-vulnerabilities-catalog",
            "https://www.cve.org/CVERecord?id=CVE-2023-4863",
            "https://www.debian.org/security/2023/dsa-5496",
            "https://www.debian.org/security/2023/dsa-5497",
            "https://www.debian.org/security/2023/dsa-5498",
            "https://www.mozilla.org/en-US/security/advisories/mfsa2023-40",
            "https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/",
            "https://www.vicarius.io/vsociety/posts/zero-day-webp-vulnerability-cve-2023-4863"
          ],
          "PublishedDate": "2023-09-12T15:15:24.327Z",
          "LastModifiedDate": "2025-03-13T16:17:15.573Z"
        },
        {
          "VulnerabilityID": "CVE-2024-28219",
          "PkgName": "Pillow",
          "PkgPath": "usr/local/lib/python3.12/site-packages/pillow-9.0.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/pillow@9.0.0",
            "UID": "41e4293c60d81b45"
          },
          "InstalledVersion": "9.0.0",
          "FixedVersion": "10.3.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-28219",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-pillow: buffer overflow in _imagingcms.c",
          "Description": "In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-680"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "bitnami": 2,
            "ghsa": 3,
            "oracle-oval": 2,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 2
          },
          "CVSS": {
            "bitnami": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 6.7
            },
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
              "V3Score": 6.7
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:L/A:H",
              "V3Score": 6.5
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:4227",
            "https://access.redhat.com/security/cve/CVE-2024-28219",
            "https://bugzilla.redhat.com/2272563",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2272563",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-28219",
            "https://errata.almalinux.org/8/ALSA-2024-4227.html",
            "https://errata.rockylinux.org/RLSA-2024:4227",
            "https://github.com/python-pillow/Pillow",
            "https://github.com/python-pillow/Pillow/commit/2a93aba5cfcf6e241ab4f9392c13e3b74032c061",
            "https://linux.oracle.com/cve/CVE-2024-28219.html",
            "https://linux.oracle.com/errata/ELSA-2024-4227.html",
            "https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-28219",
            "https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security",
            "https://ubuntu.com/security/notices/USN-6744-1",
            "https://ubuntu.com/security/notices/USN-6744-2",
            "https://ubuntu.com/security/notices/USN-6744-3",
            "https://www.cve.org/CVERecord?id=CVE-2024-28219"
          ],
          "PublishedDate": "2024-04-03T03:15:09.71Z",
          "LastModifiedDate": "2024-11-21T09:06:02.27Z"
        },
        {
          "VulnerabilityID": "CVE-2024-1135",
          "PkgName": "gunicorn",
          "PkgPath": "usr/local/lib/python3.12/site-packages/gunicorn-21.2.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/gunicorn@21.2.0",
            "UID": "5b553be5006bf19a"
          },
          "InstalledVersion": "21.2.0",
          "FixedVersion": "22.0.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:051df5c78edc2555d4810078506a0e42a69bf5c50ee9fc96fd44f0164723ef3e",
            "DiffID": "sha256:a9779c2a2f73f128d8363871f1cfd0fff1be83ba7e716ac05b97393490dc47e2"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-1135",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-gunicorn: HTTP Request Smuggling due to improper validation of Transfer-Encoding headers",
          "Description": "Gunicorn fails to properly validate Transfer-Encoding headers, leading to HTTP Request Smuggling (HRS) vulnerabilities. By crafting requests with conflicting Transfer-Encoding headers, attackers can bypass security restrictions and access restricted endpoints. This issue is due to Gunicorn's handling of Transfer-Encoding headers, where it incorrectly processes requests with multiple, conflicting Transfer-Encoding headers, treating them as chunked regardless of the final encoding specified. This vulnerability allows for a range of attacks including cache poisoning, session manipulation, and data exposure.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-444"
          ],
          "VendorSeverity": {
            "ghsa": 3,
            "redhat": 3
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:H/A:N",
              "V3Score": 8.2
            },
            "redhat": {
              "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-1135",
            "https://github.com/advisories/GHSA-w3h3-4rj7-4ph4",
            "https://github.com/benoitc/gunicorn",
            "https://github.com/benoitc/gunicorn/commit/ac29c9b0a758d21f1e0fb3b3457239e523fa9f1d",
            "https://github.com/benoitc/gunicorn/releases/tag/22.0.0",
            "https://huntr.com/bounties/22158e34-cfd5-41ad-97e0-a780773d96c1",
            "https://lists.debian.org/debian-lts-announce/2024/06/msg00027.html",
            "https://lists.debian.org/debian-lts-announce/2024/12/msg00018.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-1135",
            "https://www.cve.org/CVERecord?id=CVE-2024-1135"
          ],
          "PublishedDate": "2024-04-16T00:15:07.797Z",
          "LastModifiedDate": "2024-12-20T07:15:12.59Z"
        },
        {
          "VulnerabilityID": "CVE-2024-6827",
          "PkgName": "gunicorn",
          "PkgPath": "usr/local/lib/python3.12/site-packages/gunicorn-21.2.0.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/gunicorn@21.2.0",
            "UID": "5b553be5006bf19a"
          },
          "InstalledVersion": "21.2.0",
          "FixedVersion": "23.0.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:051df5c78edc2555d4810078506a0e42a69bf5c50ee9fc96fd44f0164723ef3e",
            "DiffID": "sha256:a9779c2a2f73f128d8363871f1cfd0fff1be83ba7e716ac05b97393490dc47e2"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-6827",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "gunicorn: HTTP Request Smuggling in benoitc/gunicorn",
          "Description": "Gunicorn version 21.2.0 does not properly validate the value of the 'Transfer-Encoding' header as specified in the RFC standards, which leads to the default fallback method of 'Content-Length,' making it vulnerable to TE.CL request smuggling. This vulnerability can lead to cache poisoning, data exposure, session manipulation, SSRF, XSS, DoS, data integrity compromise, security bypass, information leakage, and business logic abuse.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-444"
          ],
          "VendorSeverity": {
            "ghsa": 3,
            "redhat": 3
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-6827",
            "https://github.com/benoitc/gunicorn",
            "https://github.com/benoitc/gunicorn/issues/3278",
            "https://huntr.com/bounties/1b4f8f38-39da-44b6-9f98-f618639d0dd7",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-6827",
            "https://www.cve.org/CVERecord?id=CVE-2024-6827"
          ],
          "PublishedDate": "2025-03-20T10:15:33.357Z",
          "LastModifiedDate": "2025-03-20T10:15:33.357Z"
        },
        {
          "VulnerabilityID": "CVE-2021-33503",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "1.26.5",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2021-33503",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-urllib3: ReDoS in the parsing of authority part of URL",
          "Description": "An issue was discovered in urllib3 before 1.26.5. When provided with a URL containing many @ characters in the authority component, the authority regular expression exhibits catastrophic backtracking, causing a denial of service if a URL were passed as a parameter or redirected to via an HTTP redirect.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-400"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "cbl-mariner": 3,
            "ghsa": 3,
            "nvd": 3,
            "oracle-oval": 2,
            "photon": 3,
            "redhat": 2,
            "ubuntu": 1
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V2Score": 5,
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2021-33503",
            "https://errata.almalinux.org/8/ALSA-2021-4162.html",
            "https://github.com/advisories/GHSA-q2q7-5pp4-w6pg",
            "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2021-108.yaml",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/2d4a3fee6de2fa45eb82169361918f759269b4ec",
            "https://github.com/urllib3/urllib3/commit/5b047b645f5f93900d5e2fc31230848c25eb1f5f#diff-52026d639119bf1e0364836b4e8a18bd9ed3c95c6ba39b26534a5057a65e35bbR65",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-q2q7-5pp4-w6pg",
            "https://linux.oracle.com/cve/CVE-2021-33503.html",
            "https://linux.oracle.com/errata/ELSA-2021-4162.html",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/6SCV7ZNAHS3E6PBFLJGENCDRDRWRZZ6W/",
            "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FMUGWEAUYGGHTPPXT6YBD53WYXQGVV73/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6SCV7ZNAHS3E6PBFLJGENCDRDRWRZZ6W",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FMUGWEAUYGGHTPPXT6YBD53WYXQGVV73",
            "https://nvd.nist.gov/vuln/detail/CVE-2021-33503",
            "https://security.gentoo.org/glsa/202107-36",
            "https://ubuntu.com/security/notices/USN-5812-1",
            "https://www.cve.org/CVERecord?id=CVE-2021-33503",
            "https://www.oracle.com/security-alerts/cpuoct2021.html"
          ],
          "PublishedDate": "2021-06-29T11:15:07.847Z",
          "LastModifiedDate": "2024-11-21T06:08:58.03Z"
        },
        {
          "VulnerabilityID": "CVE-2023-43804",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "2.0.6, 1.26.17",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-43804",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-urllib3: Cookie request header isn't stripped during cross-origin redirects",
          "Description": "urllib3 is a user-friendly HTTP client library for Python. urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly. This issue has been patched in urllib3 version 1.26.17 or 2.0.5.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-200"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "azure": 3,
            "cbl-mariner": 2,
            "ghsa": 3,
            "nvd": 3,
            "oracle-oval": 2,
            "photon": 3,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 2
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N",
              "V3Score": 5.9
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N",
              "V3Score": 8.1
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N",
              "V3Score": 5.9
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:2159",
            "https://access.redhat.com/security/cve/CVE-2023-43804",
            "https://bugzilla.redhat.com/2242493",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2242493",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-43804",
            "https://errata.almalinux.org/9/ALSA-2024-2159.html",
            "https://errata.rockylinux.org/RLSA-2024:2986",
            "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-192.yaml",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb",
            "https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f",
            "https://linux.oracle.com/cve/CVE-2023-43804.html",
            "https://linux.oracle.com/errata/ELSA-2024-2987.html",
            "https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-43804",
            "https://security.netapp.com/advisory/ntap-20241213-0007",
            "https://security.netapp.com/advisory/ntap-20241213-0007/",
            "https://ubuntu.com/security/notices/USN-6473-1",
            "https://ubuntu.com/security/notices/USN-6473-2",
            "https://www.cve.org/CVERecord?id=CVE-2023-43804",
            "https://www.vicarius.io/vsociety/posts/cve-2023-43804-urllib3-vulnerability-3"
          ],
          "PublishedDate": "2023-10-04T17:15:10.163Z",
          "LastModifiedDate": "2024-12-13T14:15:20.57Z"
        },
        {
          "VulnerabilityID": "CVE-2020-26137",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "1.25.9",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2020-26137",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-urllib3: CRLF injection via HTTP request method",
          "Description": "urllib3 before 1.25.9 allows CRLF injection if the attacker controls the HTTP request method, as demonstrated by inserting CR and LF control characters in the first argument of putrequest(). NOTE: this is similar to CVE-2020-26116.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-74"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "cbl-mariner": 2,
            "ghsa": 2,
            "nvd": 2,
            "oracle-oval": 2,
            "photon": 2,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 2
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
              "V3Score": 6.5
            },
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:P/I:P/A:N",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
              "V2Score": 6.4,
              "V3Score": 6.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
              "V3Score": 6.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2020-26137",
            "https://bugs.python.org/issue39603",
            "https://bugzilla.redhat.com/show_bug.cgi?id=1883632",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-26137",
            "https://errata.almalinux.org/8/ALSA-2021-1761.html",
            "https://errata.rockylinux.org/RLSA-2021:1631",
            "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2020-148.yaml",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/1dd69c5c5982fae7c87a620d487c2ebf7a6b436b",
            "https://github.com/urllib3/urllib3/pull/1800",
            "https://linux.oracle.com/cve/CVE-2020-26137.html",
            "https://linux.oracle.com/errata/ELSA-2022-5235.html",
            "https://lists.debian.org/debian-lts-announce/2021/06/msg00015.html",
            "https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2020-26137",
            "https://ubuntu.com/security/notices/USN-4570-1",
            "https://usn.ubuntu.com/4570-1",
            "https://usn.ubuntu.com/4570-1/",
            "https://www.cve.org/CVERecord?id=CVE-2020-26137",
            "https://www.oracle.com/security-alerts/cpujul2022.html",
            "https://www.oracle.com/security-alerts/cpuoct2021.html"
          ],
          "PublishedDate": "2020-09-30T18:15:26.773Z",
          "LastModifiedDate": "2024-11-21T05:19:19.68Z"
        },
        {
          "VulnerabilityID": "CVE-2023-45803",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "2.0.7, 1.26.18",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-45803",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "urllib3: Request body not stripped after redirect from 303 status changes request method to GET",
          "Description": "urllib3 is a user-friendly HTTP client library for Python. urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 301, 302, or 303 after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although this behavior is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers. Because the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable. Both of the following conditions must be true to be affected by this vulnerability: 1. Using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON) and 2. The origin service is compromised and starts redirecting using 301, 302, or 303 to a malicious peer or the redirected-to service becomes compromised. This issue has been addressed in versions 1.26.18 and 2.0.7 and users are advised to update to resolve this issue. Users unable to update should disable redirects for services that aren't expecting to respond with redirects with `redirects=False` and disable automatic redirects with `redirects=False` and handle 301, 302, and 303 redirects manually by stripping the HTTP request body.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-200"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "azure": 2,
            "cbl-mariner": 2,
            "ghsa": 2,
            "nvd": 2,
            "oracle-oval": 2,
            "photon": 2,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 2
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.2
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.2
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.2
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:2132",
            "https://access.redhat.com/security/cve/CVE-2023-45803",
            "https://bugzilla.redhat.com/2246840",
            "https://bugzilla.redhat.com/2257028",
            "https://bugzilla.redhat.com/2257854",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2246840",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2257028",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2257854",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-45803",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-52323",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-22195",
            "https://errata.almalinux.org/9/ALSA-2024-2132.html",
            "https://errata.rockylinux.org/RLSA-2024:2968",
            "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-212.yaml",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/4e50fbc5db74e32cabd5ccc1ab81fc103adfe0b3",
            "https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9",
            "https://github.com/urllib3/urllib3/commit/b594c5ceaca38e1ac215f916538fb128e3526a36",
            "https://github.com/urllib3/urllib3/releases/tag/1.26.18",
            "https://github.com/urllib3/urllib3/releases/tag/2.0.7",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4",
            "https://linux.oracle.com/cve/CVE-2023-45803.html",
            "https://linux.oracle.com/errata/ELSA-2024-2988.html",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-45803",
            "https://ubuntu.com/security/notices/USN-6473-1",
            "https://ubuntu.com/security/notices/USN-6473-2",
            "https://ubuntu.com/security/notices/USN-7762-1",
            "https://www.cve.org/CVERecord?id=CVE-2023-45803",
            "https://www.rfc-editor.org/rfc/rfc9110.html#name-get"
          ],
          "PublishedDate": "2023-10-17T20:15:10.07Z",
          "LastModifiedDate": "2025-02-13T18:15:32.78Z"
        },
        {
          "VulnerabilityID": "CVE-2024-37891",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "1.26.19, 2.2.2",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-37891",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "urllib3: proxy-authorization request header is not stripped during cross-origin redirects",
          "Description": " urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-669"
          ],
          "VendorSeverity": {
            "alma": 2,
            "amazon": 2,
            "azure": 2,
            "cbl-mariner": 2,
            "ghsa": 2,
            "oracle-oval": 2,
            "redhat": 2,
            "rocky": 2,
            "ubuntu": 1
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.4
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 4.4
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2024:9458",
            "https://access.redhat.com/security/cve/CVE-2024-37891",
            "https://bugzilla.redhat.com/2292788",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2292788",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-37891",
            "https://errata.almalinux.org/9/ALSA-2024-9458.html",
            "https://errata.rockylinux.org/RLSA-2024:5309",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/40b6d1605814dd1db0a46e202d6e56f2e4c9a468",
            "https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf",
            "https://linux.oracle.com/cve/CVE-2024-37891.html",
            "https://linux.oracle.com/errata/ELSA-2024-9458.html",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-37891",
            "https://security.netapp.com/advisory/ntap-20240822-0003/",
            "https://ubuntu.com/security/notices/USN-7084-1",
            "https://ubuntu.com/security/notices/USN-7084-2",
            "https://www.cve.org/CVERecord?id=CVE-2024-37891",
            "https://www.vicarius.io/vsociety/posts/proxy-authorization-header-handling-vulnerability-in-urllib3-cve-2024-37891"
          ],
          "PublishedDate": "2024-06-17T20:15:13.45Z",
          "LastModifiedDate": "2024-11-21T09:24:28.953Z"
        },
        {
          "VulnerabilityID": "CVE-2025-50181",
          "PkgName": "urllib3",
          "PkgPath": "usr/local/lib/python3.12/site-packages/urllib3-1.25.8.dist-info/METADATA",
          "PkgIdentifier": {
            "PURL": "pkg:pypi/urllib3@1.25.8",
            "UID": "2c083c2efe4e3073"
          },
          "InstalledVersion": "1.25.8",
          "FixedVersion": "2.5.0",
          "Status": "fixed",
          "Layer": {
            "Digest": "sha256:f9e7d00d15995d12f97e54fa53d5a560139e3a0d252aac665c6e9d35c03f52f4",
            "DiffID": "sha256:85da2f49c96fb8092fc1b4cb807ea97e24ffe04e1d8630b271d73240912d653e"
          },
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2025-50181",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "urllib3: urllib3 redirects are not disabled when retries are disabled on PoolManager instantiation",
          "Description": "urllib3 is a user-friendly HTTP client library for Python. Prior to 2.5.0, it is possible to disable redirects for all requests by instantiating a PoolManager and specifying retries in a way that disable redirects. By default, requests and botocore users are not affected. An application attempting to mitigate SSRF or open redirect vulnerabilities by disabling redirects at the PoolManager level will remain vulnerable. This issue has been patched in version 2.5.0.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-601"
          ],
          "VendorSeverity": {
            "amazon": 2,
            "azure": 2,
            "ghsa": 2,
            "nvd": 2,
            "redhat": 2,
            "ubuntu": 2
          },
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 5.3
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N",
              "V3Score": 6.1
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 5.3
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2025-50181",
            "https://github.com/urllib3/urllib3",
            "https://github.com/urllib3/urllib3/commit/f05b1329126d5be6de501f9d1e3e36738bc08857",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-pq67-6m6q-mj2v",
            "https://nvd.nist.gov/vuln/detail/CVE-2025-50181",
            "https://ubuntu.com/security/notices/USN-7599-1",
            "https://ubuntu.com/security/notices/USN-7599-2",
            "https://www.cve.org/CVERecord?id=CVE-2025-50181"
          ],
          "PublishedDate": "2025-06-19T01:15:24.453Z",
          "LastModifiedDate": "2025-09-18T13:51:10.24Z"
        }
      ]
    }
  ]
}
